### PR TITLE
python3 support

### PR DESCRIPTION
**Pull request by luv**

### CODE REVIEW

**Comment by luv** [comment]
not fully tested yet
not tested with python2.7 yet

**Comment by luv** [comment]
python3: ./run_test.sh pass, commands "update" and "test" work, kerberos works

**Comment by luv** [comment]
tested with py27 as well

File: builder/commands/update.py [hunk 1]
```diff
@@ -56,9 +56,15 @@ def take_action(self, parsed_args):
 
     def parse_config(self, config_file):
         self.log.debug("Parsing the jenkins config file")
-        config = ConfigParser.ConfigParser()
+        config = configparser.ConfigParser()
         config.read(config_file)
-        user = config.get('jenkins', 'user')
-        password = config.get('jenkins', 'password')
```
Comment (hrubi) [inline builder/commands/update.py]: This was actually throwing an exception, when the option was not found, right? Why the change of the behaviour and fallback to `""`?

File: builder/commands/update.py [hunk 2]
```diff
@@ -37,7 +37,7 @@ def read_update(self, config, view_yaml):
     def take_action(self, parsed_args):
         self.log.info("Updating view data in Jenkins")
         if not parsed_args.conf:
-            print parser.print_help()
+            self.get_parser("jenkins-view-builder").print_help()
```
Comment (hrubi) [inline builder/commands/update.py]: Is this a python3 related change?

File: requirements.txt [hunk 3]
```diff
@@ -9,4 +9,3 @@ prettytable==0.7.2
 pyparsing==2.0.2
 six==1.7.3
 stevedore==0.15
-wsgiref==0.1.2
```
Comment (hrubi) [inline requirements.txt]: Unused module? Python3 related?

Comment (luv) [inline requirements.txt]: yup, for example same problem described here https://github.com/skoczen/will/issues/148

File: builder/commands/update.py (hunk 1 above)
Comment (luv) [inline builder/commands/update.py]: yeah, not python3 related, this is to support kerberos the same way jenkins-python does

File: builder/commands/update.py (hunk 2 above)
Comment (luv) [inline builder/commands/update.py]: somehow. the old expression would not even compile in py3 and we would get an error on startup

however, this is fixing it for python2 as well because parser is not in scope so it would always fail when that code would get executed

i think

**Comment by hrubi** [comment]
Please reorganize the commits - reword 4da9d81, as it changes the behaviour and it's due to kerberos login. Squash the rest. Then open this PR into upstream master and we'll merge it back to our fork/branch.
